### ISSUE TITLE
Cursor does not line up to text after () is added.

### ISSUE THREAD

**Issue by richardgobeli** [wxMaxima-developers/wxmaxima#1798]
After doing a left round bracket the cursor jumps to after the right round bracket.
This is very confusing for editing.
version from last night 9-4-2023.
I have since reloaded Aug. 24 version.

**Comment by richardgobeli** [comment]
9/9/2023 version.
![image](https://github.com/wxMaxima-developers/wxmaxima/assets/31253976/739c5058-27e6-4f84-b7b7-c4b935a1267e)
The cursor is outside the round brackets.
![image](https://github.com/wxMaxima-developers/wxmaxima/assets/31253976/fb333d2d-8586-4d3f-9125-523504082d2d)

**Comment by richardgobeli** [comment]
It is getting better, but it still has issues.
Here it shows double highlights.  The cursor is before the right round bracket.
![image](https://github.com/wxMaxima-developers/wxmaxima/assets/31253976/60a76b2f-b586-495d-acb1-9e591b9f48b0)
The other issue is it does not let me edit the line.
If I cursor left, it does not move until I do the correct number of left arrows and then it is at the beginning.  I can't select or position the cursor on the line.
Here, I did the left arrow once.
![image](https://github.com/wxMaxima-developers/wxmaxima/assets/31253976/ec150901-e840-432b-83d4-ff0e82895db3)
![image](https://github.com/wxMaxima-developers/wxmaxima/assets/31253976/2106cf09-ab15-4be0-80e9-496b8f098df0)

**Comment by gunterkoenigsmann** [comment]
Thanks for this bug report!

Perhaps this time I have tried to do a bigger change than I should have done: The crashes in the config dialogue weren't reproducible on my computer. Even valgrind didn't find out that anything misbehaved - and in the end I made a static code analysis tool run that detected several hundred places where the code could possibly be erroneous. I fixed these - and everything run finde, so I started with 3 other static code analysis tools - and the last one I tried caused me to change about 800 lines for 40 potential errors it found => getting everything to work again will probably need some time and much testing. But my hope is, still, that the result will be a much more stable program...

...hope that works out, in the end.
